### DISCUSSION TITLE
Using Python Socket.io ASGIApp with Starlette >= 0.33.0

### DISCUSSION THREAD

**Discussion by rodja** [encode/starlette#2413] [Q&A] [answered]
Up until Starlette version 0.32.0.post1 the following code works as expected:

```py
# pip install starlette==0.32.0.post1 python-socketio
import socketio
from starlette.applications import Starlette
from starlette.responses import HTMLResponse
from starlette.routing import Route

async def homepage(_):
    return HTMLResponse("""
    <!DOCTYPE html>
    <html>
    <head>
        <title>SocketIO Test</title>
        <script src="https://cdn.socket.io/4.7.3/socket.io.min.js"></script>
        <script type="text/javascript">
            document.addEventListener('DOMContentLoaded', (event) => {
                var socket = io("/", { path: "/ws/socket.io" });
                socket.on('msg', function(msg) {
                    document.body.innerHTML = '<h1>' + msg + '</h1>';
                });
            });
        </script>
    </head>
    <body>
        <h1>Connecting to SocketIO...</h1>
    </body>
    </html>
    """)

sio = socketio.AsyncServer(async_mode='asgi', cors_allowed_origins='*')
sio_app = socketio.ASGIApp(socketio_server=sio, socketio_path='socket.io')
app = Starlette(routes=[Route("/", homepage)])
app.mount('/ws', sio_app)

@sio.on('connect')
async def handle_connect(sid, *args, **kwargs):
    await sio.emit('msg', 'Hello from Starlette!')
```

After establishing the connection the website shows "Hello from Starlette!". But when upgrading to Starlette 0.33.0 or greater, the `/ws/socket.io` route is not longer found. It seems like `app.mount('/ws', sio_app)` is not mounting correctly anymore. Is there a new way of doing this? Must python-socketio's ASGIApp be changed?

**Comment by rodja** [comment]
Looks like Starlette 0.32.0.post1 passed `'/socket.io/'` as `scope['path']` to the ASGIApp `__call__` and 0.33.0+ passes `'/ws/socket.io/'`. So it seems to works when I modify the socketio_path:

```py
sio_app = socketio.ASGIApp(socketio_server=sio, socketio_path='/ws/socket.io')
```

But why has the parameter changed in 0.33.0 and was it intentional?

**Reply by Kludex** [reply]
If you try the version `0.35.1`, does it work? (We fixed some stuff regarding that)

But yeah, it was intentional.

**Reply by dreusel** [reply]
If that, being a breaking API change, was intentional then should there not have been a major version increase?

**Reply by Kludex** [reply]
> If that, being a breaking API change, was intentional then should there not have been a major version increase?

No. The change was intentional, but it's not a breaking change, it was a bug fix. Also, we don't follow SemVer yet, we follow ZeroVer (which is not important, since this would have been a minor anyway).

Please check `0.35.1`, and see if it solved the issue.

**Reply by rodja** [reply, marked as answer]
`0.35.1` still has the full path in `scope['path']`. But as I understand from https://github.com/encode/starlette/pull/2400 this is actually the ASGI standard. Therefore packages like [fastapi-socketio](https://github.com/pyropy/fastapi-socketio), [NiceGUI](https://nicegui.io) etc. which mount sub apps need to be adapted. Right?

**Reply by Kludex** [reply]
Correct.

**Reply by dreusel** [reply]
A change to an API that breaks depending code pretty much is the definition of a breaking change, right? If it also happens to fix an earlier mistake than that just also makes it a bugfix. The two are not mutually exclusive.
If you value providing a stable and reliable thing here then maybe it is time to consider upgrading to semver. But also, a fix like this should usually not be too hard to do without breaking all the code that by now has come to depend on that little accident. Offering the standards-compliant route doesn't necessarily mean that support for the oopsie-route has to be dropped, right? That's something you could always do when you're ready for that major bump.
